src/main/java/com/apigee/smartdocs/config/rest/PortalRestUtil.java

```diff
@@ -536,14 +542,32 @@ public static void postAPIDoc(ServerProfile profile, File file, boolean isCreate
   public static APIDocResponseObject getAPIDoc(ServerProfile profile, File file) throws IOException {
     HttpResponse response = null;
     try {
-	      SpecObject spec = parseSpec(profile, file);  
-	      logger.info("Getting API doc for "+ spec.getTitle());
+    	String apiIdField = profile.getApiIdField();
+	    String apiIdValue = null;
+		if (apiIdField != null && !apiIdField.equals("title")) {
```

Comment:
@nandoan - not sure I prefer the title check being hardcoded, then it doesnt make sense to have this in the pom as a property. The ideal thing would be to just look for the title field in the apicatalog json or else get the value from the pom and see if that matches any key in the apicatalog json and use that value in the code.